### PR TITLE
Archi 249 add execution context get attribute as list method

### PR DESCRIPTION
**Pull request by benoitgravitee**
## Issue

https://gravitee.atlassian.net/browse/ARCHI-249

## Description

Implementation of ExecutionContext's getAttributeAsList

## Additional context

Check the tests to make the behaviour is as excepted, take a look at https://github.com/gravitee-io/gravitee-gateway-api/pull/202

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-yqoionhobq.chromatic.com)

### CODE REVIEW

File: gravitee-apim-gateway/gravitee-apim-gateway-core/src/main/java/io/gravitee/gateway/reactive/core/context/AbstractExecutionContext.java [hunk 1]
```diff
@@ -153,6 +160,52 @@ public <T> T getAttribute(String name) {
...
+                } else {
+                    // else trim values
+                    return (List<T>) list.stream().map(String::trim).collect(Collectors.toUnmodifiableList());
+                }
+            }
+        }
+
+        if (o instanceof Collection) {
+            // copy to immutable list
+            return List.copyOf((Collection<? extends T>) o);
```
Comment (jhaeyaert) [inline gravitee-apim-gateway/gravitee-apim-gateway-core/src/main/java/io/gravitee/gateway/reactive/core/context/AbstractExecutionContext.java:191]: Probably better to rely on `Collections.unmodifiableList` which, in any case, will never create a copy but just a "read-only view".

wdyt?

**Review by jhaeyaert — APPROVED** [review]
Looks good 👍 
I've just made a small suggestion. Let you check if it makes sense or not.

File: gravitee-apim-gateway/gravitee-apim-gateway-core/src/main/java/io/gravitee/gateway/reactive/core/context/AbstractExecutionContext.java (hunk 1 above)
Comment (benoitgravitee) [reply gravitee-apim-gateway/gravitee-apim-gateway-core/src/main/java/io/gravitee/gateway/reactive/core/context/AbstractExecutionContext.java:191]: Better indeed.

Comment (benoitgravitee) [reply gravitee-apim-gateway/gravitee-apim-gateway-core/src/main/java/io/gravitee/gateway/reactive/core/context/AbstractExecutionContext.java:191]: In fact, no... it requires a list, I can use unmodifiableCollection, but I'll have to convert to a list incase it is another collection type

**Pull request merged**